gator/checks/check_CountCommits.py

```diff
@@ -0,0 +1,71 @@
+"""Check that a repository has the required number of commits."""
+
+import argparse
+
+from gator import checkers
+from gator import constants
+from gator import invoke
+
+# import snoop
+# snoop.install(color="rrt")
+
+
+def get_parser():
+    """Get a parser for the arguments provided on the command-line."""
+    # create the parser with the default help formatter
+    # use a new description since this is a stand-alone check
+    parser = argparse.ArgumentParser(
+        prog="CountCommits",
+        description="Check Provided by GatorGrader: CountCommits",
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+    )
+
+    # Required Named Checker Arguments {{{
+
+    required_group = parser.add_argument_group("required check arguments")
+
+    # COUNT: the number of commits
+    # REQUIRED? Yes
+    required_group.add_argument(
+        "--count", type=int, help="minimum number of git commits", required=True
+    )
+
+    # }}}
+
+    # Optional Named Checker Arguments {{{
+
+    optional_group = parser.add_argument_group("optional check arguments")
+
+    # EXACT: perform exact checking for commit counts (i.e,. "==" instead of ">=")
+    # REQUIRED? No
+    optional_group.add_argument(
```

Comment:
I've thought about this idea and I'm still unsure about whether or not it is a good idea. My current thinking is that these `check_` modules should be self-documenting and self-contained, even at the expense of code duplication, but with the exception of the `act` function.

I suggest that we wait to tackle this issue until another PR. We should think carefully about questions such as how plugins can call external functions &mdash; should they be in a separate module just for the shared code among the checks? should they be in the main `gator` package? I think that it would be best to have plugin-shared code (e.g., for the arguments) in a separate package from the existing `gator` package. What do you think? Is it okay to defer this to a new issue for further discussion and then implementation?

Reply:
I agree on all points -- probably best to have a separate package, in a different PR.